gems/pending/disk/modules/VhdxDisk.rb

```diff
@@ -373,9 +374,13 @@ def region_table_entry(table_offset, count)
     VHDX_REGION_TABLE_ENTRY.decode(@vhdx_file.read(SIZEOF_VHDX_REGION_TABLE_ENTRY))
   end
 
+  def bat(entry)
+    @bat.empty? && process_bat
+    @bat[entry]
+  end
+
```

Comment:
Typically, when you have a dependency chain like that, it pays to investigate memoization of the other parts of the chain. For example, maybe we should add a `total_bat_entries` method.